packages/core/src/entrypoint/index.ts

```diff
@@ -0,0 +1,190 @@
+import EntryPoint_v6 from "./0.6.js";
+import EntryPoint_v7 from "./0.7.js";
+import type {
+  DefaultEntryPointVersion,
+  EntryPointDef,
+  EntryPointDefRegistry,
+  EntryPointRegistry,
+  EntryPointVersion,
+  GetEntryPointOptions,
+  SupportedEntryPoint,
+} from "./types.js";
+
+export const defaultEntryPointVersion: DefaultEntryPointVersion = "0.6.0";
+
+export const entryPointRegistry: EntryPointRegistry = {
+  "0.6.0": EntryPoint_v6,
+  "0.7.0": EntryPoint_v7,
+};
+
+export const coerceEntryPointVersion = <
+  TEntryPointVersion extends EntryPointVersion
+>(
+  entryPoint: EntryPointRegistry[EntryPointVersion],
+  userOperation: UserOperationLike,
+  throwable?: boolean
+): UserOperationLike<TEntryPointVersion> | undefined => {
+  if (entryPoint.isUserOpVersion(userOperation)) {
+    return userOperation as UserOperationLike<TEntryPointVersion>;
+  }
+  if (!throwable) {
+    return undefined;
+  }
+  throw new MismatchingEntryPointError(entryPoint.version, userOperation);
+};
+
+export function getEntryPoint<
+  TEntryPointVersion extends EntryPointVersion = DefaultEntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[TEntryPointVersion];
+
+export function getEntryPoint<
+  TEntryPointVersion extends DefaultEntryPointVersion = DefaultEntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options?: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[TEntryPointVersion];
+
+export function getEntryPoint<TChain extends Chain = Chain>(
+  chain: TChain,
+  options?: GetEntryPointOptions<DefaultEntryPointVersion>
+): EntryPointDefRegistry<TChain>[DefaultEntryPointVersion];
+
+export function getEntryPoint<
+  TEntryPointVersion extends EntryPointVersion,
+  TChain extends Chain = Chain
+>(
+  chain: TChain,
+  options: GetEntryPointOptions<TEntryPointVersion>
+): EntryPointDefRegistry<TChain>[EntryPointVersion] {
+  const { version = defaultEntryPointVersion, addressOverride } = options ?? {
+    version: defaultEntryPointVersion,
+  };
+
+  const entryPoint = entryPointRegistry[version ?? defaultEntryPointVersion];
+  const address = addressOverride || entryPoint.address[chain.id];
+  if (!address) {
+    throw new EntryPointNotFoundError(chain, version);
+  }
+
+  if (version === "0.6.0") {
+    const v6 = entryPoint as SupportedEntryPoint<
+      "0.6.0",
+      TChain,
+      typeof EntryPointAbi_v6
+    >;
+    return {
+      version: v6.version,
+      address,
+      chain,
+      abi: v6.abi,
+      getUserOperationHash: (r) =>
+        v6.getUserOperationHash(r, address, chain.id),
+      packUserOperation: v6.packUserOperation,
+      isUserOpVersion: v6.isUserOpVersion,
+      coerce: (userOperation, throwable) =>
+        coerceEntryPointVersion(v6, userOperation, throwable),
+    } as EntryPointDef<"0.6.0", TChain, typeof EntryPointAbi_v6>;
```

Comment:
better we use `satisfies` here instead of `as` for some additional type safety

Reply:
can't use satisfies here. typescript errors.

Reply:
see below, we should fix the typescript errors that keeps satisfies from working

Reply:
see https://github.com/alchemyplatform/aa-sdk/pull/549

the reason satisfies was not working is because the resulting object doesn't contain an `overrides` property which is required (though can be set to undefined explicitly) we should fix that.